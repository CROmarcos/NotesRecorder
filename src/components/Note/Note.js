import './Note.scss'
import heart from '../../../src/images/heart.png'
import trashCan from '../../../src/images/trashcan.png'
import pen from '../../../src/images/pen.png'
import emptyHeart from '../../../src/images/emptyheart.png'
import printer from '../../../src/images/printer.png'
import alarm from '../../../src/images/alarm.png'
import accept from '../../../src/images/accept.png'
import cancel from '../../../src/images/cancel.png'
import store from '../../state/store'
import { useState } from "react"
import { deleteNote, markNote } from '../../state/actions'
import { Link } from 'react-router-dom'
import { setAlarm } from '../../state/actions'
import AlarmNotify, { getTimeString, openAlarm, makeAlarm, cancelAlarm } from '../Alarm/alarm'

const Note = props => {

    const [notification, setNotification] = useState("N/A")

    let hours = []
    let minutes = []

    for (let i = 0; i < 60; i++) {
        if (i < 24) hours.push(i)
        let min = i
        if (i < 10) min = "0" + min
        minutes.push(min)
    }

    const saveAlarm = (id) => {
        let selectedHours = document.getElementById(`hours/${id}`)
        let hr = selectedHours.options[selectedHours.selectedIndex].value
        let selectedMinutes = document.getElementById(`minutes/${id}`)
        let min = selectedMinutes.options[selectedMinutes.selectedIndex].value
        store.dispatch(setAlarm({ id, alarm: getTimeString({ hours: hr, minutes: min, seconds: "00" }) }))
        setNotification(getTimeString({ hours: hr, minutes: min, seconds: "00" }))
    }

    const deleteAlarm = (id) => {
        store.dispatch(setAlarm({ id, alarm: "N/A" }))
    }

    return (
        <div className='card'>
            <div className='card__title'>
                <section className='card__title-text'>
                    <p>{props.title}</p>
                </section>
                <section className='card__title-actions'>
                    <Link to={`/edit/${props.noteId}`}><button className='action'><img src={pen} alt="edit" /></button></Link>
                    <button className='action' onClick={() => { store.dispatch(deleteNote(props.noteId)); props.refresh() }}><img src={trashCan} alt="delete"></img></button>
                    <button style={props.filtered ? { display: "none" } : { display: "flex" }} data-testid="mark-note" className='action' onClick={() => { store.dispatch(markNote({ id: props.noteId, importance: !props.important })); props.refresh() }}>{props.important ? <img src={heart} alt="important note" /> : <img src={emptyHeart} alt="important note" />}</button>
                </section>
            </div>
            <div className='card__body'>
                <div className='card__body-row'><label>Task: </label><span>{props.description}</span></div>
                <div className='card__body-row'><label>Created: </label><span>{props.creationDate}</span></div>
                <div className='card__body-row'><label>Last modified: </label><span>{props.lastModification}</span></div>
            </div>
            <div className='card__footer'>
                <section className='card__footer--side left'>
                    <div id={`open-alarm/${props.noteId}`} className='alarm--open'>
                        <button className='action' onClick={() => openAlarm(props.noteId)}><img src={alarm} alt="Make notification" /></button>
                    </div>
                    <div id={`set-alarm/${props.noteId}`} className='alarm--set'>
                        <label>hrs&nbsp;</label>
                        <select id={`hours/${props.noteId}`}>
                            {hours.map(hr => <option key={hr}>{hr}</option>)}
                        </select>
                        <label>&nbsp;min&nbsp;</label>
                        <select id={`minutes/${props.noteId}`}>
                            {minutes.map(min => <option key={min}>{min}</option>)}
                        </select>
                        &emsp;
                        <button className='action small' onClick={() => { saveAlarm(props.noteId); AlarmNotify(props.noteId); makeAlarm(props.noteId) }}><img src={accept} alt="Set alarm" /></button>
                        &emsp;
                        <button className='action small' onClick={() => cancelAlarm(props.noteId)}><img src={cancel} alt="Cancel alarm" /></button>
                    </div>
                    <div id={`active-alarm/${props.noteId}`} className='alarm'>
                        <p>Alarm set for {notification}</p>&emsp;
                        <button className='action small' onClick={() => { deleteAlarm(props.noteId); cancelAlarm(props.noteId) }}><img src={cancel} alt="Cancel alarm" /></button>
                    </div>
                </section>
                <section className='card__footer--side'>
                    <Link to={`/print/${props.noteId}`}><button className='action'><img src={printer} alt="Save to PDF" /></button></Link>
                </section>
            </div>
        </div>
    )
}

export default Note