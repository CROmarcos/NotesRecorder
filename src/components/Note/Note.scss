.card {
    width: 400px;
    min-height: 250px;
    border: 2px solid rgb(2, 2, 59);
    border-radius: 2%;
    margin-bottom: 50px;

    &__title {
        width: 100%;
        height: 20%;
        border-bottom: 2px solid rgb(2, 2, 59);
        display: inline-flex;

        &-text {
            width: 60%;
            height: 100%;
            display: flex;
            justify-content: left;
            align-items: center;
            padding-left: 15px;
            font-size: 14pt;
            font-weight: bold;
        }

        &-important {
            width: 40%;
            height: 100%;
            display: flex;
            justify-content: space-evenly;
            align-items: center;
        }
    }

    &__body {
        width: calc(100% - 20px);
        height: calc(60% - 10px);
        padding: 5px 10px;
        font-size: 12pt;
        text-align: left;

        &-row {
            width: 100%;
            margin-bottom: 10px;

            label {
                font-weight: bold;
            }
        }
    }

    &__footer {
        width: calc(100% - 15px);
        height: 20%;
        display: flex;
        justify-content: right;
        padding-right: 15px;
    }
}

.action {
    width: fit-content;
    height: 35px;
    background: none;
    border: none;
    font-weight: bold;
    text-decoration: underline;
    cursor: pointer;
    margin: 0;
    padding: 0;

    img {
        height: 100%;
    }
}
