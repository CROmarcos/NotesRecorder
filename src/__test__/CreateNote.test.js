import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import Note from "../components/Note/Note"

it("note matches snapshot", () => {
    const card = renderer.create(
        <MemoryRouter>
            <Note id="9999" title="Test note" description="Note made for testing" creationDate="today" lastModification="N/A" filtered={false} />
        </MemoryRouter>
    ).toJSON()
    expect(card).toMatchSnapshot()
})