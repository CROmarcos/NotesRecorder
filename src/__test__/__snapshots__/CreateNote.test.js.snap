// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`note matches snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__title"
  >
    <section
      className="card__title-text"
    >
      <p>
        Test note
      </p>
    </section>
    <section
      className="card__title-actions"
    >
      <a
        href="/edit/undefined"
        onClick={[Function]}
      >
        <button
          className="action"
        >
          <img
            alt="edit"
            src="pen.png"
          />
        </button>
      </a>
      <button
        className="action"
        onClick={[Function]}
      >
        <img
          alt="delete"
          src="trashcan.png"
        />
      </button>
      <button
        className="action"
        data-testid="mark-note"
        onClick={[Function]}
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <img
          alt="important note"
          src="emptyheart.png"
        />
      </button>
    </section>
  </div>
  <div
    className="card__body"
  >
    <div
      className="card__body-row"
    >
      <label>
        Task: 
      </label>
      <span>
        Note made for testing
      </span>
    </div>
    <div
      className="card__body-row"
    >
      <label>
        Created: 
      </label>
      <span>
        today
      </span>
    </div>
    <div
      className="card__body-row"
    >
      <label>
        Last modified: 
      </label>
      <span>
        N/A
      </span>
    </div>
  </div>
  <div
    className="card__footer"
  >
    <section
      className="card__footer--side left"
    >
      <div
        className="alarm--open"
        id="open-alarm/undefined"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <button
          className="action"
          onClick={[Function]}
        >
          <img
            alt="Make notification"
            src="alarm.png"
          />
        </button>
      </div>
      <div
        className="alarm--set"
        id="set-alarm/undefined"
      >
        <label>
          hrs 
        </label>
        <select
          id="hours/undefined"
        >
          <option>
            0
          </option>
          <option>
            1
          </option>
          <option>
            2
          </option>
          <option>
            3
          </option>
          <option>
            4
          </option>
          <option>
            5
          </option>
          <option>
            6
          </option>
          <option>
            7
          </option>
          <option>
            8
          </option>
          <option>
            9
          </option>
          <option>
            10
          </option>
          <option>
            11
          </option>
          <option>
            12
          </option>
          <option>
            13
          </option>
          <option>
            14
          </option>
          <option>
            15
          </option>
          <option>
            16
          </option>
          <option>
            17
          </option>
          <option>
            18
          </option>
          <option>
            19
          </option>
          <option>
            20
          </option>
          <option>
            21
          </option>
          <option>
            22
          </option>
          <option>
            23
          </option>
        </select>
        <label>
           min 
        </label>
        <select
          id="minutes/undefined"
        >
          <option>
            00
          </option>
          <option>
            01
          </option>
          <option>
            02
          </option>
          <option>
            03
          </option>
          <option>
            04
          </option>
          <option>
            05
          </option>
          <option>
            06
          </option>
          <option>
            07
          </option>
          <option>
            08
          </option>
          <option>
            09
          </option>
          <option>
            10
          </option>
          <option>
            11
          </option>
          <option>
            12
          </option>
          <option>
            13
          </option>
          <option>
            14
          </option>
          <option>
            15
          </option>
          <option>
            16
          </option>
          <option>
            17
          </option>
          <option>
            18
          </option>
          <option>
            19
          </option>
          <option>
            20
          </option>
          <option>
            21
          </option>
          <option>
            22
          </option>
          <option>
            23
          </option>
          <option>
            24
          </option>
          <option>
            25
          </option>
          <option>
            26
          </option>
          <option>
            27
          </option>
          <option>
            28
          </option>
          <option>
            29
          </option>
          <option>
            30
          </option>
          <option>
            31
          </option>
          <option>
            32
          </option>
          <option>
            33
          </option>
          <option>
            34
          </option>
          <option>
            35
          </option>
          <option>
            36
          </option>
          <option>
            37
          </option>
          <option>
            38
          </option>
          <option>
            39
          </option>
          <option>
            40
          </option>
          <option>
            41
          </option>
          <option>
            42
          </option>
          <option>
            43
          </option>
          <option>
            44
          </option>
          <option>
            45
          </option>
          <option>
            46
          </option>
          <option>
            47
          </option>
          <option>
            48
          </option>
          <option>
            49
          </option>
          <option>
            50
          </option>
          <option>
            51
          </option>
          <option>
            52
          </option>
          <option>
            53
          </option>
          <option>
            54
          </option>
          <option>
            55
          </option>
          <option>
            56
          </option>
          <option>
            57
          </option>
          <option>
            58
          </option>
          <option>
            59
          </option>
        </select>
         
        <button
          className="action small"
          onClick={[Function]}
        >
          <img
            alt="Set alarm"
            src="accept.png"
          />
        </button>
         
        <button
          className="action small"
          onClick={[Function]}
        >
          <img
            alt="Cancel alarm"
            src="cancel.png"
          />
        </button>
      </div>
      <div
        className="alarm"
        id="active-alarm/undefined"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <p>
          Alarm set for 
          21:00
        </p>
         
        <button
          className="action small"
          onClick={[Function]}
        >
          <img
            alt="Cancel alarm"
            src="cancel.png"
          />
        </button>
      </div>
    </section>
    <section
      className="card__footer--side"
    >
      <a
        href="/print/undefined"
        onClick={[Function]}
      >
        <button
          className="action"
        >
          <img
            alt="Save to PDF"
            src="printer.png"
          />
        </button>
      </a>
    </section>
  </div>
</div>
`;
